examples/two-columns-layout/src/app.tsx

```diff
@@ -28,7 +31,44 @@ autocomplete({
     productsPlugin,
     articlesPlugin,
     popularPlugin,
+    quickAccessPlugin,
   ],
+  reshape({ sourcesBySourceId }) {
+    const {
+      recentSearchesPlugin: recentSearches,
+      querySuggestionsPlugin: querySuggestions,
+      categoriesPlugin: categories,
+      brandsPlugin: brands,
+      faqPlugin: faq,
+      ...rest
+    } = sourcesBySourceId;
+
+    const removeDuplicates = uniqBy(({ source, item }) => {
+      if (
+        ['recentSearchesPlugin', 'querySuggestionsPlugin'].indexOf(
+          source.sourceId
+        ) === -1
+      ) {
+        return item;
+      }
+
+      return source.sourceId === 'querySuggestionsPlugin'
+        ? item.query
+        : item.label;
+    });
+
+    const fillWith = populate({
+      mainSourceId: 'querySuggestionsPlugin',
+      limit: 8,
+    });
+
+    const combine = pipe(removeDuplicates, fillWith);
```

Comment:
these functions can be defined outside of the function body, right?

Reply:
Yes, should I place them in the same file, at the beginning or in an other file?

Reply:
I think top-level of this file makes most sense, to avoid more indirection of the reshape

Reply:
Thanks!